src/backend/joseki.be/webapp/Controllers/AuditsController.cs

```diff
@@ -15,32 +16,54 @@ namespace webapp.Controllers
     [Route("api/audits")]
     public class AuditsController : Controller
     {
-    /// <summary>
-    /// Returns the overall infrastructure overview for Joseki landing page.
-    /// </summary>
-    /// <returns>The overall infrastructure overview.</returns>
+        private static List<InfrastructureComponentSummary> overallSummary = Data.GetComponentSummary();
+        private static List<InfrastructureComponentSummary> componentSummaries = Data.GetComponentSummaries();
+
+        /// <summary>
+        /// Returns the overall infrastructure overview for Joseki landing page.
+        /// </summary>
+        /// <returns>The overall infrastructure overview.</returns>
         [HttpGet]
         [Route("overview", Name = "get-overview")]
         [ProducesResponseType(200, Type = typeof(InfrastructureOverview))]
         public Task<ObjectResult> GetOverview(DateTime? date = null)
         {
-            // a simple mechanism to lookup the overall data from Data date dictionary
-            // it should use the data layer instead.
             var summary = new InfrastructureOverview();
-            var indexDate = Data.Overall.Keys.First();
-            if (date != null)
+            var indexDate = (date == null) ? Data.Dates.First() : date;
+
+            summary.Overall = overallSummary.FirstOrDefault(summary => summary.Date == indexDate);
+            summary.Components = componentSummaries.Where(summary => summary.Date == indexDate).ToArray();
+
+            return Task.FromResult(this.StatusCode(200, summary));
+        }
+
+        /// <summary>
+        /// Returns the overall infrastructure overview diff.
+        /// </summary>
+        /// <returns>The overall infrastructure overview diff.</returns>
+        [HttpGet]
+        [Route("overviewdiff", Name = "get-overview-diff")]
```

Comment:
What about a route `/api/overview/diff?date1=20200101&date2=20200102`, where `date1` and `date2` are query parameters? I think, it's more REST-friendly

Reply:
makes sense! route changed to `/api/overview/diff`